scipy/stats/_binned_statistic.py

```diff
@@ -664,3 +632,70 @@ def binned_statistic_dd(sample, values, statistic='mean',
     result = result.reshape(input_shape[:-1] + list(nbin-2))
 
     return BinnedStatisticddResult(result, edges, binnumbers)
+
+
+def _bin_edges(sample, bins=None, range=None):
+
+    Dlen, Ndim = sample.shape
+
+    nbin = np.empty(Ndim, int)    # Number of bins in each dimension
+    edges = Ndim * [None]         # Bin edges for each dim (will be 2D array)
+    dedges = Ndim * [None]        # Spacing between edges (will be 2D array)
+
+    # Select range for each dimension
+    # Used only if number of bins is given.
+    if range is None:
+        smin = np.atleast_1d(np.array(sample.min(axis=0), float))
```

Comment:
this e.g. converts `int` input types into `float`. I realize this was behavior from before but if the float part is dropped here I think we would retain the type in the numpy array

Reply:
I do not understand.
Feel free to change it.

Reply:
I guess the `smin` and `smax` are being used to determine bidedges which we do want to be floats so lets leave as is.